src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8075,23 +8075,32 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
             }
 
             bool argIsIndex = argIsIndexNotRange.Value();
-            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
-            if (!TryFindIndexOrRangeIndexerPattern(receiverOpt, receiverType, argIsIndex: argIsIndex,
-                out PropertySymbol? lengthOrCountProperty, out Symbol? patternSymbol, diagnostics, ref useSiteInfo))
+            var receiverValEscape = receiverOpt switch
+            {
+                BoundListPatternReceiverPlaceholder { ValEscape: var valEscape } => valEscape,
+                BoundSlicePatternReceiverPlaceholder { ValEscape: var valEscape } => valEscape,
+                _ => GetValEscape(receiverOpt, LocalScopeDepth)
```

Comment:
>GetValEscape(receiverOpt, LocalScopeDepth) [](http://example.com/codeflow?start=21&length=42)

Can we simplify code, always do that and get rid of the switch?

Reply:
Indeed. Thanks